src/fraction/unicode_fromto_str.rs

```diff
@@ -0,0 +1,589 @@
+use std::fmt::Display;
+
+use crate::fraction::GenericFraction;
+use crate::{error::ParseError, Sign};
+use num::rational::Ratio;
+use num::Zero;
+use std::{fmt, str};
+use Integer;
+
+impl<T: Clone + Integer + Display + From<u8>> GenericFraction<T> {
+    /// Display the fraction using FRACTION SLASH '\u{2044}' '⁄'
+    /// If you have font support, this is the way Unicode wants to display fractions
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).display_unicode()),
+    ///   "1⁄2"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).display_unicode()),
+    ///   "3⁄2"
+    /// );
+    /// ```
+    pub fn display_unicode(&self) -> impl fmt::Display + '_ {
+        struct UnicodeDisplay<'a, T: Clone + Integer>(&'a GenericFraction<T>);
+        impl<'a, T> fmt::Display for UnicodeDisplay<'a, T>
+        where
+            T: Clone + Integer + fmt::Display,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                match &self.0 {
+                    GenericFraction::NaN => write!(f, "NaN"),
+                    GenericFraction::Infinity(s) => {
+                        write!(f, "{}∞", s)
+                    }
+                    GenericFraction::Rational(s, r) => {
+                        write!(f, "{}{}", s, r.numer())?;
+                        if !r.denom().is_one() {
+                            write!(f, "⁄{}", r.denom())?;
+                        }
+                        Ok(())
+                    }
+                }
+            }
+        }
+        UnicodeDisplay(self)
+    }
+
+    /// Display the fraction using FRACTION SLASH '\u{2044}' '⁄' as a mixed fraction e.g. "1⁤1⁄2"
+    /// Will put INVISIBLE PLUS '\u{2064}' as a separator '⁤'
+    /// If you have font support, this is the way Unicode wants to display fractions
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).display_unicode_mixed()),
+    ///   "1⁄2"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).display_unicode_mixed()),
+    ///   "1⁤1⁄2"
+    /// );
+    /// ```
+    pub fn display_unicode_mixed(&self) -> impl Display + '_ {
+        struct S<'a, T: Clone + Integer + fmt::Display>(&'a GenericFraction<T>);
```

Comment:
I'd really love us to have a public `UnicodeDisplay` struct, which could be _configurable_. Thus, instead of having extra methods on the GenericFraction API surface, we could do something neat like `format!("{}", fraction.get_unicode_display().mixed())`? We could later expand the `UnicodeDisplay` own API with more relevant configurational/behavioral methods.

Reply:
Yeah, the reason I implemented it like this (actually I also would've liked it to be on the other struct) is that FromStr trait is def'd as:

```
Impl FromStr for T {
  fn from_string(s &str) -> Self {

  }
}
```

And I want a Fraction as output, not the display struct. I could also make a trait or something that allows for `fn from_str(s &str) -> Fraction`.

Reply:
Ah no, heres an idea! We have one function: `write!("{} glass of milk", f.get_unicode_display(DisplayType::FSlash))`

I really dont like long function names, hencewhy I didnt add 'get' but can do :)

Reply:
get_fancy_display :D

Reply:
Whoops I misread your comment, thats actually better! (Bit weird though, since then we would have an impl display impl display, but thats fine)